clippy_lints/src/loops/missing_spin_loop.rs

```diff
@@ -0,0 +1,33 @@
+use super::MISSING_SPIN_LOOP;
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::is_no_std_crate;
+use rustc_errors::Applicability;
+use rustc_hir::{Block, Expr, ExprKind};
+use rustc_lint::LateContext;
+//use rustc_middle::ty;
+use rustc_span::sym;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, cond: &'tcx Expr<'_>, body: &'tcx Expr<'_>) {
+    if_chain! {
+        if let ExprKind::Block(Block { stmts: [], expr: None, ..}, _) = body.kind;
+        if let ExprKind::MethodCall(method, _, [_callee, _ordering], _) = cond.kind;
+        if method.ident.name == sym::load;
+        //if let ty::Adt(def, _substs) = cx.typeck_results().expr_ty(callee).kind();
+        //if cx.tcx.is_diagnostic_item(sym::AtomicBool, def.did);
```

Comment:
Without this, this will lint on all calls to `load` on an empty loop. I guess this is fine, because empty loops are probably rare and regardless of the type `load` is called on, this is suspicious and probably warrants a `hint::spin_loop()`.

Please remove those lines and possibly replace them with a comment.

Reply:
I'd actually like to check this, but I think it fails because `AtomicBool` is missing the `rustc_diagnostic_item` attribute. I've opened a PR to add it.

Reply:
Doesn't this lint also apply to `AtomicU8` and all the integer types? Or do you plan to check for all of them?